Adding a blog to discuss release cadence change

jeremyrickard opened the pull request:
Adding a blog discussing the release cadence changes brought about by the [release cadence kep](https://github.com/kubernetes/enhancements/tree/master/keps/sig-release/2572-release-cadence), in advance of the 1.23 release (which will be the last one for the year).

jeremyrickard commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+slug: new-kubernetes-release-cadence
+---
+
+REF: https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/
```
Need to remove this

jeremyrickard replied:
removed

saschagrunert commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+
+## What's changing and when
+
+Starting with the [Kubernetes 1.22 release](https://www.kubernetes.dev/resources/release/), a lightweight policy will drive the creation of each release schedule. This policy states:
```
https://www.kubernetes.dev/resources/release/ will link to 1.23 if it's going to be released, right? If so, can we find a link which always points to 1.22?

palnabarun replied:
We can use https://github.com/kubernetes/sig-release/tree/master/releases/release-1.22

jeremyrickard replied:
updated to use the sig-release repo link from @palnabarun

saschagrunert commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+* An explicit SIG Release break of at least two weeks between each cycle will
+   be enforced.
+
+As a result, Kubernetes will follow a three release pear year cadence. Kubernetes 1.23 will be the final release of the 2021 calendar year. This new schedule creation policy results in a very predictible release schedule, allowing us to forecast upcoming release dates:
```
"schedule" is used twice within the sentence, let's find a synonym for one of both.

chrisnegus replied:
Change "three release pear year" to "three release per year"

jeremyrickard replied:
done!

saschagrunert commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+
+*Proposed Kubernetes Release Schedule for the remainder of 2021*
+
+| Week Number in Year | Release Number | Release Week | Note |
+| -------- | -------- | -------- | -------- |
+| 35 | 3 | 1 (August 23) | | 
+| 50 | 3 | 16 (December 07) | KubeCon + CloudNativeCon NA Break (Oct 11-15) | 
+
+*Proposed Kubernetes Release Schedule for 2022*
+
+| Week Number in Year | Release Number | Release Week | Note |
+| -------- | -------- | -------- | -------- |
+| 1  | 1 | 1 (January 03) | |
+| 15 | 1 | 15 (April 12) | | 
+| 17 | 2 | 1 (April 26) | KubeCon + CloudNativeCon EU likely to occur |
+| 32 | 2 | 15 (August 09) | |
+| 34 | 3 | 1 (August 22 | KubeCon + CloudNativeCon NA likely to occur |
+| 49 | 3 | 14 (December 06) |
```
I'd suggest exchanging the "Release Number" with the actual release version. How about putting it in one table to make everything more compact, like:

| Calendar Weeks | Release | Timespan | Duration in Weeks | Note |
| -------- | -------- | -------- | - | -------- |
| 35-50 | v1.22 | Aug 23 - Dec 07 | 16 | KubeCon NA Break (Oct 11-15) |
| 1-15 | v1.23 | Jan 3 - Apr 12 | 15 | |
| 17-32 | v1.24 | Apr 26 - Aug 9 | 15 | KubeCon EU likely to occur |
| 34-49 | v1.25 | Aug 22 - Dec 6 | 14 | KubeCon NA likely to occur |

jeremyrickard replied:
done!

saschagrunert commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+| 34 | 3 | 1 (August 22 | KubeCon + CloudNativeCon NA likely to occur |
+| 49 | 3 | 14 (December 06) |
+
+These proposed dates reflect only the start and end dates, and they are subject to change. The Release Team will select dates for enhancement freeze, code freeze, and other milestones at the start of each release. Feedback from prior releases will feed into this process.
```
Let's link the releases phases here: https://github.com/kubernetes/sig-release/blob/master/releases/release_phases.md

sftim replied:
Could also link via https://www.k8s.dev/resources/release/#phases if that looks better.

jeremyrickard replied:
done!

saschagrunert commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+
+The only major change end users will experience is a slower release cadence. Kubernetes release artifacts, release notes, and all other aspects of any given release will stay the same.
+
+With less releases to consume per year, end user organizations will spend less time on upgrades and more time on supporting their Kubernetes clusters. It also means that Kubernetes releases are in support for a slightly longer period of time.
```
Not sure if we can speak for end users, should we make it a bit more intentional?
```suggestion
With less releases to consume per year, it's intended that end user organizations will spend less time on upgrades and gain more time on supporting their Kubernetes clusters. It also means that Kubernetes releases are in support for a slightly longer period of time.
```

chrisnegus replied:
I believe it is more correct to say "fewer releases" than "less releases".

saschagrunert commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+
+## Why we decided to change the release cadence
+
+The Kubernetes 1.19 cycle was far longer than usual. SIG Release extended it to lessen the burden on both Kubernetes contributors and Kubernetes end users due the COVID-19 pandemic. Following this extended release, the Kubernetes 1.20 release became the third, and final, release for 2020. 
```
```suggestion
The Kubernetes 1.19 cycle was far longer than usual. SIG Release extended it to lessen the burden on both Kubernetes contributors and end users due the COVID-19 pandemic. Following this extended release, the Kubernetes 1.20 release became the third, and final, release for 2020. 
```

sftim replied:
Do we know how much the COVID-19 aspect is correct? Anecdotally, the folks who commented to me about having 3 releases per year were pleased because it lowered the effort for teams managing clusters, and none mentioned the pandemic.

jeremyrickard replied:
We 100% made the decision because of Covid-19 originally, then further extended it because of other events that unfolded during the year. It was not explicitly done to have fewer releases so folks would have less upgrade, but having less upgrades as a side effect is true.

justaugustus commented:
/assign @saschagrunert @justaugustus

chrisnegus commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+
+As the Kubernetes project matures, the number of enhancements per cycle grows, along with the burden on contributors, the Release Engineering team.  Downstream consumers and integrators also face increased challenges keeping up with [ever more feature-packed releases](https://kubernetes.io/blog/2021/04/08/kubernetes-1-21-release-announcement/). A wider project adoption means the complexity of sup ...
+
+Changing the release cadence from four to three releases per year balances a variety of factors for stakeholders: while it's not strictly an LTS policy, consumers and integrators will get longer support terms for each minor version as the extended release cycles lead to the [last three releases being supported](https://kubernetes.io/blog/2020/08/31/kubernetes-1-19-feature-one-year-support/) for a ...
```
Perhaps change "last three releases being supported" to "previous three releases being supported"

jeremyrickard replied:
done.

chrisnegus commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+
+These proposed dates reflect only the start and end dates, and they are subject to change. The Release Team will select dates for enhancement freeze, code freeze, and other milestones at the start of each release. Feedback from prior releases will feed into this process.
+
+## What this means for end users 
```
You don't mention any downside for changing the release cadence. If there is none, then that's fine. But I'm wondering if there will be some impact of fixes or enhancements having to wait for a longer period of time. Is there something built into the new cadence that can mitigate those potential negative effects?

jeremyrickard replied:
I can add this, I think we addressed it in the KEP so I'll paraphrase that content here, if that works?

chrisnegus replied:
Yes. That was the only thing I wondered after reading the whole document.

jeremyrickard replied:
The negatives really come down to slower feature graduation and velocity. There isn't much that can be mitigated, it's just a result of the current policy around graduation (number of releases) and time. I've added information for users to be aware of the impact.

pmmalinov01 commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+---
+layout: blog
+title: "Kuberentes Release Cadence Change: Here’s What You Need To Know"
+date: 2021-07-01
```
This date is a bit in the past, Do we know a future date that this blog post will be out? I do not want to put pressure on you, I want to reference this blog post in the Major themes for the 1.22 release notes.

sftim replied:
https://kubernetes.io/docs/contribute/new-content/blogs-case-studies/ has some more context on this (but could be more clear). The blog team can pick a date and update it to be in the future, so long as [maintainer pushes](https://kubernetes.io/docs/contribute/review/for-approvers/#commit-into-another-person-s-pr) to the source fork are permitted.

sftim replied:
The first challenge here is finding folks to do the first-pass review; once that's happened, I think it'll be easier to get a date set.

sftim commented on content/en/blog/_posts/2021-07-12-Kubernetes-Release-Cadence/index.md:
```diff
@@ -0,0 +1,84 @@
+---
+layout: blog
+title: "Kuberentes Release Cadence Change: Here’s What You Need To Know"
```
```suggestion
title: "Kubernetes Release Cadence Change: Here’s What You Need To Know"
```

jeremyrickard commented:
@justaugustus @saschagrunert @sftim @chrisnegus I think I've addressed all comments so far, could you take another pass? @sftim any other reviewers you can suggest?

saschagrunert reviewed (approved):
/lgtm

cpanato reviewed (approved):
/lgtm

onlydole commented:
@jeremyrickard - this looks good to me! The final piece would be to update the date to tomorrow or Wednesday this week and we can approve (barring any other feedback that arises). 🙌🏼

onlydole reviewed (approved):
Thank you, @jeremyrickard and team! 

/lgtm
